apps/hash-agents/app/agents/generate-entity-type-property-types/__main__.py

```diff
@@ -0,0 +1,75 @@
+import json
+
+from langchain.chat_models import ChatOpenAI
+from langchain.schema import HumanMessage, SystemMessage
+
+from .io_types import Input, Output, PropertyTypeDefinition
+
+SYSTEM_MESSAGE_CONTENT = " ".join(
+    [
+        "You are an entity type generator.",
+        (
+            "Given a title and description of an entity type, generate relevant"
+            " property types for the entity."
+        ),
+        "Respond to any user prompts in a minified JSON array format (no whitespace),",
+        "where each item in the array represents a property type for the entity type",
+        "where each property type has a `title`, `description`,",
+        # TODO - support additional data types
+        "and a `dataType` (one of `text`, `number` or `boolean`).",
+    ]
+)
+
+
+def main(agent_input: Input) -> Output:
+    # TODO - add support for querying existing property types so
+    # that they can be re-used
+    chat = ChatOpenAI(model_name="gpt-3.5-turbo", temperature=0)
+
+    messages = [
+        SystemMessage(content=SYSTEM_MESSAGE_CONTENT),
+        HumanMessage(
+            content=json.dumps(
+                {
+                    "title": agent_input.entity_type_title,
+                    "description": agent_input.entity_type_description,
+                }
+            )
+        ),
+    ]
+
+    response = chat(messages)
+
+    property_type_definitions = json.loads(response.content)
```

Comment:
I have a TODO a few lines down for validating the response, where we can perhaps look into feeding validation errors back into the chat or handling them in some other way. If that's what you're referring to, I can expand on the todo.

I'm not sure whether it's worth investing into as part of this PR, as I don't think I faced any issues with the usage `json.loads` here. And if an error were to occur, the follow-up error handling PR would let the user retry the request.

Reply:
It's just it would have been easy to miss this later on. Your TODO about improving error handling and such was below this. So I was asking us to make it clearer we need to handle this later on with either a try except or a TODO on this line  specifically.

Reply:
So you could commit the error you have here if you like, _and_ add a TODO to improve it with retry logic / whatever

Reply:
Thanks

Reply:
Understood, I've added `try ... catch` with a custom error message in https://github.com/hashintel/hash/pull/2486/commits/2295dd4dd49a5acfce06b91e1f31269cdb4e29fb, and expanded on the TODOs on how we could improve the handling of errors that might occur